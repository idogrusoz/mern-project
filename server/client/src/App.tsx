import { MuiThemeProvider } from "@material-ui/core/styles";
import React from "react";
import { RouteComponentProps } from "react-router-dom";
import "./App.css";
import { AuthContextProvider } from "./components/auth/AuthContext";
import NavBar from "./components/NavBar";
import Routes from "./components/Routes";
import theme from "./theme";

function App(props: RouteComponentProps<{}>) {
    return (
        <AuthContextProvider>
            <MuiThemeProvider theme={theme}>
                <NavBar />
                <div style={{ marginTop: 70 }}>
                    <Routes {...props} />
                </div>
            </MuiThemeProvider>
        </AuthContextProvider>
    );
}

export default App;
