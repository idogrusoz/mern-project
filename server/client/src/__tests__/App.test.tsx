import React from "react";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { MemoryRouter } from "react-router-dom";
import * as ReactRouterDom from "react-router-dom";
import App from "../App";

test("matches the snapshot", () => {
    const location = { ...window.location, state: "/" };
    const history = createMemoryHistory();
    const match: ReactRouterDom.match = {
        isExact: true,
        params: {},
        path: "/",
        url: "",
    };
    const app = render(<App history={history} location={location} match={match} />, { wrapper: MemoryRouter });
    expect(app).toMatchSnapshot();
});
