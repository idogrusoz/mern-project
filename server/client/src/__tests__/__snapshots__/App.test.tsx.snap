// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          style="display: flex; justify-content: space-around;"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
            style="font-weight: 900;"
          >
            KERPICH
          </h6>
        </div>
      </header>
      <div
        style="margin-top: 70px;"
      >
        <div
          style="height: 100%; display: flex; justify-content: center;"
        >
          <div
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        style="display: flex; justify-content: space-around;"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
          style="font-weight: 900;"
        >
          KERPICH
        </h6>
      </div>
    </header>
    <div
      style="margin-top: 70px;"
    >
      <div
        style="height: 100%; display: flex; justify-content: center;"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
